pkg/registry/authentication/tokenreview/storage.go

```diff
@@ -90,6 +90,18 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 		fakeReq = fakeReq.WithContext(authenticator.WithAudiences(fakeReq.Context(), auds))
```

Comment:
```suggestion
		fakeReq = fakeReq.WithContext(authenticator.WithAudiences(ctx, auds))
```

Why not this?

Reply:
are there other things in the context we don't want to mix with the evaluation, like audit event annotations?

Reply:
Right, I was avoiding the whole "decide what to do with audit annotations when authentication runs twice" problem for now.  I plan to open a PR for that later, but in general we need to be careful with flows that mutate the context under the assumption that they only run once with the same set of parameters (audit annotations being the only example I am aware of).

Reply:
No, I don't think so. Audit annotations (at the version of the PR that I lgtm'd) are only supported in incoming request filters and are otherwise ignored so they do not have an issue here. This is fundamentally the same request context (just like any other registry function), so if passing the original context here is an issue, the bug is in the context handling.

Reply:
Setting audit annotations during REST storage is explicitly supported, even without my PR.  My concern here is:

1. You have auth method 1 that sets audit annotation `A=B`
2. You have auth method 2 that sets audit annotation `C=D`
3. Token review comes in using method 1, sets `A=B`
4. Internal rest call invokes method 2, sets `C=D`
5. `defer` in `WithAudit` runs and has both `A=B` and `C=D` set

Reply:
I think we should solve that in audit. Options are:
* Separate methods for SetOverride and SetIfNotAnnotationForKey.
* Some heuristic like first annotation for key wins.
Given that the registry doesn't run in a detached context, I don't think we should solve this by constructing an artificial context here.

Reply:
> Separate methods for SetOverride and SetIfNotAnnotationForKey.

Not sure what you mean here.

> Some heuristic like first annotation for key wins.

We already have that today for when the keys used are exactly the same.  But what is expected output when different types of authenticators are used?  To me it seems weird to just merge them.  For example, there could be overlap between some of the keys and not others, so you get some weird state in the middle.  Also, if the keys are exactly the same and first wins, you are silently dropping information, which does not seem great.

Reply:
It's not a problem now, but I would like to eventually forward context keys used for tracing. If we have attributes that don't play nicely, I'm thinking that a list of context attributes that we should not forward is going to be longterm easier to maintain than a list of context attributes to forward.